meilisearch-http/src/index_controller/dump_actor/actor.rs

```diff
@@ -44,144 +43,93 @@ where
-        let uuids: HashSet<(String, Uuid)> = uuids.into_iter().collect();
+    async fn handle_message(&self, msg: DumpMsg) {
+        use DumpMsg::*;
 
-        if uuids.is_empty() {
-            return Ok(());
+        match msg {
+            CreateDump { ret } => {
+                let _ = self.handle_create_dump(ret).await;
+            }
+            DumpInfo { ret, uid } => {
+                let _ = ret.send(self.handle_dump_info(uid).await);
+            }
         }
+    }
 
-        let indexes = self.list_indexes().await?;
-
-        // we create one directory by index
-        for meta in indexes.iter() {
-            tokio::fs::create_dir(temp_dump_path.join(&meta.uid)).await?;
+    async fn handle_create_dump(&self, ret: oneshot::Sender<DumpResult<DumpInfo>>) {
+        if self.is_running().await {
+            ret.send(Err(DumpError::DumpAlreadyRunning))
+                .expect("Dump actor is dead");
+            return;
         }
+        let uid = generate_uid();
+        let info = DumpInfo::new(uid.clone(), DumpStatus::InProgress);
+        *self.dump_info.write().await = Some(info.clone());
 
-        let metadata = super::Metadata::new(indexes, env!("CARGO_PKG_VERSION").to_string());
-        metadata.to_path(&temp_dump_path).await?;
-
-        self.update.dump(uuids, temp_dump_path.clone()).await?;
-
-        let dump_dir = self.dump_path.clone();
-        let dump_path = self.dump_path.join(format!("{}.dump", uid));
-        let dump_path = tokio::task::spawn_blocking(move || -> anyhow::Result<PathBuf> {
-            let temp_dump_file = tempfile::NamedTempFile::new_in(dump_dir)?;
-            let temp_dump_file_path = temp_dump_file.path().to_owned();
-            compression::to_tar_gz(temp_dump_path, temp_dump_file_path)?;
-            temp_dump_file.persist(&dump_path)?;
-            Ok(dump_path)
-        })
-        .await??;
-
-        info!("Created dump in {:?}.", dump_path);
+        ret.send(Ok(info)).expect("Dump actor is dead");
 
-        Ok(())
-    }
+        let dump_info = self.dump_info.clone();
 
-    async fn list_indexes(&self) -> anyhow::Result<Vec<IndexMetadata>> {
-        let uuids = self.uuid_resolver.list().await?;
+        let task_result = tokio::task::spawn(perform_dump(
+            self.dump_path.clone(),
+            self.uuid_resolver.clone(),
+            self.index.clone(),
+            self.update.clone(),
+            uid.clone(),
+        ))
+        .await;
 
-        let mut ret = Vec::new();
-
-        for (uid, uuid) in uuids {
-            let meta = self.index.get_index_meta(uuid).await?;
-            let meta = IndexMetadata {
-                uuid,
-                name: uid.clone(),
-                uid,
-                meta,
-            };
-            ret.push(meta);
-        }
-
-        Ok(ret)
+        match task_result {
+            Ok(Ok(())) => {
+                (*dump_info.write().await).as_mut().expect("Dump actor should have an inbox").done();
```

Comment:
Uhm, the  error is unrelated with the inbox, the error message should be something like "Inconsistent dump service state"